src/main/java/core/basesyntax/service/impl/TransactionParserImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransactionParserImpl implements TransactionParser {
+    private static final String SEPARATOR = ",";
+    private static final int EXPECTED_PARTS = 3;
+    private static final int OPERATION_TYPE = 0;
+    private static final int FRUIT_NAME = 1;
+    private static final int QUANTITY = 2;
+
+    @Override
+    public List<Transaction> parse(List<String> rawData) {
+        if (rawData == null || rawData.isEmpty()) {
+            throw new IllegalArgumentException("Invalid transaction line: " + rawData);
+        }
```

Comment:
According to the checklist, we should avoid beginning method implementation with an empty line. Please remove the empty line at the start of the `parse` method.